[PATCH] using pkgconfig to check libxml

Upstream-Status: Pending

Use pkg-config for the libxml2 dependency, not the -config script.

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
---
 acinclude.m4 | 48 ++++++------------------------------------------
 1 file changed, 6 insertions(+), 42 deletions(-)

diff --git a/acinclude.m4 b/acinclude.m4
index d348f57..9f691a5 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -2530,49 +2530,13 @@ dnl
 dnl Common setup macro for libxml
 dnl
 AC_DEFUN([PHP_SETUP_LIBXML], [
-AC_CACHE_CHECK([for xml2-config path], ac_cv_php_xml2_config_path,
-[
-  for i in $PHP_LIBXML_DIR /usr/local /usr; do
-    if test -x "$i/bin/xml2-config"; then
-      ac_cv_php_xml2_config_path="$i/bin/xml2-config"
-      break
-    fi
-  done
-])
+  PKG_CHECK_MODULES(PKG_XML2, [libxml-2.0],,)
+  if test "x$PKG_XML2_CFLAGS" != "x"; then
+    PHP_EVAL_INCLINE($PKG_XML2_CFLAGS, $1)
+    PHP_EVAL_LIBLINE($PKG_XML2_LIBS)
+    AC_DEFINE(HAVE_LIBXML, 1, [ ])
+    $2
 
-  if test -x "$ac_cv_php_xml2_config_path"; then
-    XML2_CONFIG="$ac_cv_php_xml2_config_path"
-    libxml_full_version=`$XML2_CONFIG --version`
-    ac_IFS=$IFS
-    IFS="."
-    set $libxml_full_version
-    IFS=$ac_IFS
-    LIBXML_VERSION=`expr [$]1 \* 1000000 + [$]2 \* 1000 + [$]3`
-    if test "$LIBXML_VERSION" -ge "2006011"; then
-      LIBXML_LIBS=`$XML2_CONFIG --libs`
-      LIBXML_INCS=`$XML2_CONFIG --cflags`
-      PHP_EVAL_LIBLINE($LIBXML_LIBS, $1)
-      PHP_EVAL_INCLINE($LIBXML_INCS)
-
-      dnl Check that build works with given libs
-      AC_CACHE_CHECK(whether libxml build works, php_cv_libxml_build_works, [
-        PHP_TEST_BUILD(xmlInitParser,
-        [
-          php_cv_libxml_build_works=yes
-        ], [
-          AC_MSG_RESULT(no)
-          AC_MSG_ERROR([build test failed.  Please check the config.log for details.])
-        ], [
-          [$]$1
-        ])
-      ])
-      if test "$php_cv_libxml_build_works" = "yes"; then
-        AC_DEFINE(HAVE_LIBXML, 1, [ ])
-      fi
-      $2
-    else
-      AC_MSG_ERROR([libxml2 version 2.6.11 or greater required.])
-    fi
 ifelse([$3],[],,[else $3])
   fi
 ])
-- 
1.9.1

